### PR TITLE
Issue #11720: Kill surviving mutation in PackageObjectFactory

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
#11720

### Diff Reports: 
N/A

Missing Test Case Was Added

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java [hunk 1]
```diff
@@ -491,4 +528,8 @@ protected void processFiltered(File file, FileText fileText) {
 
     }
 
+    private static class MockClass {
+        MockClass() {
```
Comment (nick-mancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java:532]: Do we need explicit default constructor?

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
Item:

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java (hunk 1 above)
Comment (Vyom-Yadav) [reply src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java:532]: If the class has an access modifier `private` then the default constructor by default also has a `private` access modifier, reflection can find that constructor (we won't get `NoSuchMethodException`) but we will get `IllegalAccessException`. So if we use private then the default constructor either needs to be `public` or with the default access modifier.

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java [hunk 2]
```diff
@@ -478,6 +479,42 @@ public void testGenerateThirdPartyNameToFullModuleNameWithException() throws Exc
...
+    }
+
+    @Test
+    public void testCreateModuleWithTryInAllRegisteredPackages() {
+        final ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        final Set<String> packages = Collections.singleton(BASE_PACKAGE);
+        final PackageObjectFactory objectFactory =
+            new PackageObjectFactory(packages, classLoader, SEARCH_REGISTERED_PACKAGES);
+        try {
+            objectFactory.createModule("PackageObjectFactoryTest$MockClass");
```
Comment (nick-mancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java:502]: Please use assertThrows.

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
Item:

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java (hunk 2 above)
Comment (Vyom-Yadav) [reply src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java:503]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java (hunk 1 above)
Comment (Vyom-Yadav) [reply src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java]: @nick-mancuso Semaphore is complaining, `setAccessable(true)` won't work as I would have to change the code in `PackageObjectFactory` to do so.

Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java]: https://github.com/checkstyle/checkstyle/blob/399f1b48b79731d65e824eb927971ef6174c3463/src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java#L96

Comment (nick-mancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java]: Let’s skip this, if constructor is not redundant, I am good.

**Review by nick-mancuso — APPROVED** [review]

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
@Vyom-Yadav original implementation of was OK, let’s make CI happy so we can merge this.

**Comment by Vyom-Yadav** [comment]
@nick-mancuso Done, semaphore was giving `The constructor PackageObjectFactoryTest.MockClass() is never used locally` so I changed the access modifier of the class so we don't have to deal with constructor.

**Review by nick-mancuso — APPROVED** [review]

**Comment by nick-mancuso** [comment]
Build failure in Travis and no-error-xwiki are not related to changes in this PR.

**Pull request merged**